frontend/src/pages/Form/Form.tsx

```diff
@@ -0,0 +1,249 @@
+import React, { useEffect, useState } from "react";
+import { useFormik } from "formik";
+import { validation } from "./validation";
+import {
+  StyledBottomWrapper,
+  StyledButtonWrapper,
+  StyledSubmitButton,
+  StyledCancelButton,
+  StyledTopWrapper,
+  StyledContainer,
+  StyledTitle,
+  StyledForm,
+} from "./Form.styled";
+import { IForm } from "./Form.model";
+import { useRouteMatch, useHistory } from "react-router-dom";
+import { getUserObject, isObjectEmpty } from "helpers";
+import { ISocial, IUser } from "Types/interfaces";
+import Input from "components/atoms/Input";
+import Checkbox from "components/atoms/Checkbox";
+import SocialList from "components/molecules/SocialList";
+
+import {
+  createUser,
+  getAllSocials,
+  getAllUsers,
+  getUser,
+  updateUser,
+} from "Services";
+
+const Form = ({ edit }: IForm): JSX.Element => {
+  const history = useHistory();
+  const { params } = useRouteMatch<{ id: string }>();
+  const [initialValues, setInitialValues] = useState(getUserObject());
+  const [users, setUsers] = useState<IUser[]>([]);
+  const [socials, setSocials] = useState<ISocial[]>([]);
+
+  const {
+    values,
+    errors,
+    touched,
+    handleBlur,
+    handleChange,
+    handleSubmit,
+  } = useFormik({
+    validationSchema: validation(),
+    enableReinitialize: true,
+    validateOnChange: true,
+    initialValues,
+    onSubmit: (values) => {
+      console.log(values)
+      onSubmit(values, !!edit);
+    },
+  });
+
+  useEffect(() => {
+    getAllUsers().then(({ data }) => {
+      setUsers(data);
+    });
+  }, [setUsers]);
+
+  useEffect(() => {
+    getAllSocials().then(({ data }) => {
+      setSocials(data);
+    });
+  }, [setSocials]);
+
+  useEffect(() => {
+   console.log(errors, values)
+  }, [errors, values]);
+
+  const onCancel = () => history.replace("/admin");
+  const onSubmit = (data: IUser, edit: boolean) => {
+    const {
+      name,
+      boss,
+      image,
+      title,
+      active,
+      socials,
+      manager,
+      openToWork,
+    } = data;
+
+    edit && updateUser(data).then(console.log);
+    !edit &&
+      createUser({
+        name,
+        boss,
+        image,
+        title,
+        active,
+        socials,
+        manager,
+        openToWork,
+      });
+    setTimeout(() => {
+      onCancel();
+    }, 100);
+  };
```

Comment:
why do you double check variables edit? How working this part of code? This is the same below?
setTimeout(() => {
      onCancel();
    }, 100);
 if(edit) {
   updateUser(data).then(console.log);
   return;
 }
createUser({
        name,
        boss,
        image,
        title,
        active,
        socials,
        manager,
        openToWork,
      });
   
Which approach is better? 
Again, when is catch error?